src/packet.ml

```diff
@@ -319,7 +322,16 @@ let encode_tls_data t =
         Cstruct.concat
           [ u_l; Cstruct.of_string u; null; p_l; Cstruct.of_string p; null ]
   and opt = Cstruct.of_string t.options in
-  Cstruct.concat [ prefix; key_source; opt_len; opt; null; u_p ]
+  let peer_info = match t.peer_info with
+    | None -> Cstruct.empty
+    | Some peer_info ->
+      let len = String.length peer_info in
+      let pi = Cstruct.create (2 + len) in
+      Cstruct.BE.set_uint16 pi 0 len;
+      Cstruct.blit_from_string peer_info 0 pi 0 len;
+      pi
```

Comment:
we need an additional 0 byte, i.e. 2 byte length, then peer info, then null

Reply:
see #122 for a helper... you also blit peer_info at the beginning, while it should be at offset 2 (since it's [n_0; n_1; data_0; ..; data_n; 0]